### PR TITLE
Add code-actions to generate payload and header param template

### PR DESCRIPTION
**Pull request by TharmiganK**
## Purpose

> $Subject

> Fixes [`Add code-actions to generate payload and header param template #2642`](https://github.com/ballerina-platform/ballerina-standard-library/issues/2642)

Introducing two code-actions to add payload and header parameter to the resource signature :
1. `Add payload parameter` -> will add `@http:Payload json payload` to the resource signature
2. `Add header parameter` -> will add `@http:Header string header` to the resource signature

## Example
Feb-09-2022 14-18-30

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpResourceValidator.java [hunk 1]
```diff
@@ -449,6 +455,23 @@ public static void extractInputParamTypeAndValidate(SyntaxNodeAnalysisContext ct
...
+        if (resourceMethodOptional.isPresent() && !payloadAnnotationPresent) {
+            enableAddPayloadParamCodeAction(ctx, member.location(), resourceMethodOptional.get());
+        }
+        if (!headerAnnotationPresent) {
+            enableAddHeaderParamCodeAction(ctx, member.location());
+        }
+    }
+
+    private static void enableAddPayloadParamCodeAction(SyntaxNodeAnalysisContext ctx, Location location,
+                                                       String methodName) {
```
Comment (chamil321) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpResourceValidator.java:467]:
```suggestion
                                                        String methodName) {
```

File: compiler-plugin-tests/src/test/resources/ballerina_sources/sample_codeaction_package_2/service.bal [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+service /greeting on new http:Listener(0) {
+    resource function get hi () returns string {
```
Comment (chamil321) [inline compiler-plugin-tests/src/test/resources/ballerina_sources/sample_codeaction_package_2/service.bal:20]:
```suggestion
    resource function get hi() returns string {
```

Comment (chamil321) [reply compiler-plugin-tests/src/test/resources/ballerina_sources/sample_codeaction_package_2/service.bal:20]: Let's remove other spaces as well

File: compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddAnnotatedParam.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import java.util.List;
+import java.util.Optional;
+
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.FUNCTION_SIGNATURE;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.RESOURCE_ACCESSOR_DEFINITION;
+
+/**
+ * Abstract implementation of code action to add an annotated parameter to resource signature.
+ */
+public abstract class AddAnnotatedParam implements CodeAction {
```
Comment (chamil321) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddAnnotatedParam.java:51]: The class name should be a noun. Apparently previously added classes also starts with a verb. Is that the common practice for CodeAction classes?

File: compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpResourceValidator.java (hunk 1 above)
Comment (TharmiganK) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpResourceValidator.java]: Fixed in [`239e4bd`](https://github.com/ballerina-platform/module-ballerina-http/pull/880/commits/239e4bdc40cf610cabdae97329c50c88c8a30f59)

File: compiler-plugin-tests/src/test/resources/ballerina_sources/sample_codeaction_package_2/service.bal (hunk 2 above)
Comment (TharmiganK) [reply compiler-plugin-tests/src/test/resources/ballerina_sources/sample_codeaction_package_2/service.bal]: Fixed in [`239e4bd`](https://github.com/ballerina-platform/module-ballerina-http/pull/880/commits/239e4bdc40cf610cabdae97329c50c88c8a30f59)

File: compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddAnnotatedParam.java (hunk 3 above)
Comment (TharmiganK) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddAnnotatedParam.java:51]: In [language-server code actions](https://github.com/ballerina-platform/ballerina-lang/tree/master/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers), the class names starts like a verb and end with `CodeAction` (For example `AddAnnotatedParameterCodeAction`)
Shall we follow the same convention?

Comment (chamil321) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddAnnotatedParam.java:51]: +1

Comment (TharmiganK) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddAnnotatedParam.java]: Fixed in [0c4163a](https://github.com/ballerina-platform/module-ballerina-http/pull/880/commits/0c4163a781d4b39592292f3c4b0fbf10b2b469f7)

**Review by chamil321 — APPROVED** [review]

**Pull request merged**